site/app/models/gradeable/Gradeable.php

```diff
@@ -2593,4 +2513,419 @@ public function hasLeaderboard(): bool {
         }
         return !empty($autograding_config->getLeaderboards());
     }
+
+    public function getId(): string {
+        return $this->id;
+    }
+
+    public function getTitle(): string {
+        return $this->title;
+    }
+
+    public function getInstructionsUrl(): string {
+        return $this->instructions_url;
+    }
+
+    public function setInstructionsUrl(string $url): void {
+        $this->instructions_url = $url;
+        $this->modified = true;
```

Comment:
Is setting `$modified` part of why magic methods are useful for this class?  I don't like the idea of having to set this variable in every setter, present and future.

Reply:
I agree, and I think that we should not be tracking `$modified` as it is infrequently accessed and I don't believe there is a code path that checks it when it could be false.